src/http/client.cc

```diff
@@ -92,19 +92,14 @@ future<> connection::write_body(const request& req) {
 }
 
 future<connection::reply_ptr> connection::maybe_wait_for_continue(const request& req) {
-    if (req.get_header("Expect") == "") {
-        return make_ready_future<reply_ptr>(nullptr);
+    if (req.get_header("Expect") != "") {
+        co_await _write_buf.flush();
```

Comment:
The common case is, that Expect will be empty, no? so we add an allocation here.

Please measure the performance of the series.

Reply:
> The common case is, that Expect will be empty, no? so we add an allocation here.

The allocation is unavoidable here anyway, at the end this method does `co_await recv_reply()` and it will wait for it

> Please measure the performance of the series.

ok

Reply:
> Please measure the performance of the series.

With #2442 , the test makes 1M requests in
* master: 3.40 +/- 0,02 seconds (min 3,37 / max 3,50)
* this PR: 3,48 +/- 0,02 seconds (min 3,45 / max 3,62)

Reply:
Please report allocs/op

Reply:
* master: 45 allocs/req
* this PR: 46 allocs/req